src/components/grid/GridComponent.vue

```diff
@@ -17,17 +18,20 @@
             </th>
           </tr>
         </table>
+
         <div :class="{'space-holder':stickyTableHeader || !grid}"></div>
-        <div class="table-holder">
+
+        <div class="table-holder" v-if="isLoading || grid && grid.length">
           <loading
             :active="isLoading"
             loader="dots"
             :is-full-page="false"
             color="var(--secondary)"
             background-color="var(--light)"
           ></loading>
+
           <table
-            v-if="grid"
+            v-if="grid && grid.length && !isLoading"
```

Comment:
looks like grid && grid.length is checked in parent at line 24

Reply:
Fixed